### PR TITLE
Chore: UI - Replace Block Escaping with Escaper

### PR DESCRIPTION
**Pull request by pykettk**
### Description (*)
Refactors the `Magento_Ui` module to replace `$block` escaping functions with `$escaper` escaping functions.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [x] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#37095: Chore: Admin Analytics - Replace Block Escaping with Escaper

### CODE REVIEW

**Comment by pykettk** [comment]
@magento create issue

**Review by fredden — APPROVED** [review]

**Comment by engcom-Lima** [comment]
@magento run all tests

File: app/code/Magento/Ui/view/base/templates/control/button/default.phtml [hunk 1]
```diff
@@ -3,13 +3,15 @@
...
  */
+declare(strict_types=1);
 
-/**
- * @var $block \Magento\Ui\Component\Control\Button
- */
+use Magento\Framework\Escaper;
+
+/** @var Escaper $escaper */
+/** @var \Magento\Ui\Component\Control\Button $block */
```
Comment (leonhelmus) [inline app/code/Magento/Ui/view/base/templates/control/button/default.phtml:11]: maybe import the class

File: app/code/Magento/Ui/view/base/templates/stepswizard.phtml [hunk 2]
```diff
@@ -62,7 +68,7 @@
         "*": {
             "Magento_Ui/js/core/app": {
                 "components": {
-                        "<?= $block->escapeJs($block->getComponentName()) ?>": {
+                        "<?= $escaper->escapeJs($block->getComponentName()) ?>": {
                             "component": "Magento_Ui/js/lib/step-wizard",
                             "initData": <?= /* @noEscape */ $this->helper(Magento\Framework\Json\Helper\Data::class)->jsonEncode($block->getInitData()) ?>,
```
Comment (leonhelmus) [inline app/code/Magento/Ui/view/base/templates/stepswizard.phtml:73]: Does this even work? I think it misses a backslash at the beginning.
```
Magento\Framework\Json\Helper\Data::class
```

**Review by leonhelmus — APPROVED** [review]
Added 2 comments nothing blocking.

File: app/code/Magento/Ui/view/base/templates/stepswizard.phtml (hunk 2 above)
Comment (fredden) [reply app/code/Magento/Ui/view/base/templates/stepswizard.phtml:73]: It'd work both ways. If there is a leading backslash, then the path is absolute. Without a leading backslash, then the path is relative to the current namespace (or imported names via `use` statements). Given there is no namespace declared in this file, the default namespace of `\` is implied.

**Pull request closed without merging**